qubit-sdk/src/main/java/com/qubit/android/sdk/internal/placement/interactor/PlacementInteractorImpl.kt

```diff
@@ -0,0 +1,79 @@
+package com.qubit.android.sdk.internal.placement.interactor
+
+import com.qubit.android.sdk.api.tracker.OnPlacementError
+import com.qubit.android.sdk.api.tracker.OnPlacementSuccess
+import com.qubit.android.sdk.internal.configuration.repository.ConfigurationModel
+import com.qubit.android.sdk.internal.configuration.repository.ConfigurationRepository
+import com.qubit.android.sdk.internal.placement.PlacementImpl
+import com.qubit.android.sdk.internal.placement.callback.PlacementCallbackConnectorImpl
+import com.qubit.android.sdk.internal.placement.connector.PlacementAPI
+import com.qubit.android.sdk.internal.placement.connector.PlacementConnector
+import com.qubit.android.sdk.internal.placement.connector.PlacementConnectorImpl
+import com.qubit.android.sdk.internal.placement.model.PlacementMode
+import com.qubit.android.sdk.internal.placement.model.PlacementModel
+import com.qubit.android.sdk.internal.placement.model.PlacementPreviewOptions
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+internal class PlacementInteractorImpl(
+    private val configurationRepository: ConfigurationRepository,
+    private val deviceId: String
+) : PlacementInteractor {
+
+  companion object {
+    private val DEFAULT_PLACEMENT_MODE = PlacementMode.LIVE
+    private const val BASE_URL_PLACEHOLDER = "http://localhost/"
+  }
+
+  override fun fetchPlacement(
+      placementId: String,
+      mode: PlacementMode?,
+      previewOptions: PlacementPreviewOptions,
+      onSuccess: OnPlacementSuccess,
+      onError: OnPlacementError
+  ) {
+    buildConnector().getPlacementModel(
+        getPlacementApiHost(configurationRepository),
+        placementId,
+        mode ?: DEFAULT_PLACEMENT_MODE,
+        deviceId,
+        previewOptions,
+        { handleSuccessfulResponse(it, onSuccess, onError) },
+        { onError(it) }
+    )
+  }
+
+  private fun buildConnector(): PlacementConnector {
+    val retrofit = Retrofit.Builder()
+        .baseUrl(BASE_URL_PLACEHOLDER)  // https://stackoverflow.com/questions/34842390/how-to-setup-retrofit-with-no-baseurl
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+    val placementAPI = retrofit.create(PlacementAPI::class.java)
+
+    return PlacementConnectorImpl(placementAPI)
```

Comment:
I don't like PlacementConnectorImpl is created here. And why it is created on every fetchPlacement call. It seems to be constant.
Class SDK is intented to build components and joining them together.
I think, PlacementConnector can be dependency of this class injected by constructor.
PlacementAPI can be created inside PlacementConnectorImpl as it is internal part of PlacementConnector. It could be also created in SDK.java and injected as dependency to PlacementConnectorImpl, however I think it would be overengineering.

For sure this is not good place for creating PlacementAPI.

Reply:
Moved that logic to separate `PlacementConnectorBuilder` class as that's how we do it in case of other `*Connector` classes. `SDK` doesn't create them explicitly, but always does it through builder classes.